tests/integration_test.go

```diff
@@ -60,12 +60,13 @@ func (s *IntegrationTestSuite) SetupTest() {
 		// We need slightly more than xmrmakerProvideAmount for transaction fees
 		mineMinXMRMakerBalance(s.T(), common.MoneroToPiconero(xmrmakerProvideAmount*2))
 	}
-	s.setSwapTimeout(defaultSwapTimeout) // reset between tests, so every test starts in a known state
-}
 
-func (s *IntegrationTestSuite) setSwapTimeout(timeoutSeconds uint64) {
-	ac := rpcclient.NewClient(defaultXMRTakerSwapdEndpoint)
-	err := ac.SetSwapTimeout(timeoutSeconds)
+	// Reset XMR Maker and Taker between tests, so tests starts in a known state
+	ac := rpcclient.NewClient(context.Background(), defaultXMRTakerSwapdEndpoint)
+	err := ac.SetSwapTimeout(defaultSwapTimeout)
+	require.NoError(s.T(), err)
+	bc := rpcclient.NewClient(context.Background(), defaultXMRMakerSwapdEndpoint)
+	err = bc.ClearOffers(nil)
```

Comment:
Adding context handling to rpcclient broke a bunch of integration tests, because they were calling `bc.ClearOffers(nil)` in a defer statement that executed after the context that `bc` was created with was cancelled. I removed the cleanup in the individual tests in favor of putting the swapd instances into a know state before each test starts here.